vendor/k8s.io/kubernetes/openshift-kube-apiserver/admission/network/apis/externalipranger/register.go

```diff
@@ -18,3 +28,5 @@ func addKnownTypes(scheme *runtime.Scheme) error {
 	)
 	return nil
 }
+
+func (obj *ExternalIPRangerAdmissionConfig) GetObjectKind() schema.ObjectKind { return &obj.TypeMeta }
```

Comment:
didn't expect this to be necessary

Reply:
These functions are used in the call tree while registering the openshift kube admission plugins. If they aren't present, the type registration fails. The reason I stumbled upon this was because I was trying to plug a hole in 4.1 (https://github.com/openshift/cluster-kube-apiserver-operator/pull/714) where we allow external IPs to be configured on a service by default (since we don't have a field in the network spec for "ExternalIP"). While doing so, I figured that the `ExternalIPRangerAdmissionConfig` was not getting registered as a type because the above methods were missing. I figured this needed to be fixed all the way from 4.1 to master. So I have all the backport PRs ready. The fix is truly needed for 4.1 however.

Reply:
Type registration failures:

```E0115 03:44:12.096883       1 helpers.go:126] Encountered config error no kind "ExternalIPRangerAdmissionConfig" is registered for version "v1" in scheme "github.com/openshift/origin/pkg/cmd/server/apis/config/register.go:12" in object *externalipranger.ExternalIPRangerAdmissionConfig, raw JSON:
{"apiVersion":"network.openshift.io/v1","kind":"ExternalIPRangerAdmissionConfig"}
E0115 03:44:12.096952       1 helpers.go:126] Encountered config error no kind "RestrictedEndpointsAdmissionConfig" is registered for version "v1" in scheme "github.com/openshift/origin/pkg/cmd/server/apis/config/register.go:12" in object *restrictedendpoints.RestrictedEndpointsAdmissionConfig, raw JSON:
{"apiVersion":"network.openshift.io/v1","kind":"RestrictedEndpointsAdmissionConfig","restrictedCIDRs":["10.128.0.0/14","172.30.0.0/16"]}```

https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-kube-apiserver-operator/714/pull-ci-openshift-cluster-kube-apiserver-operator-release-4.1-e2e-aws/45/artifacts/e2e-aws/pods/openshift-kube-apiserver_kube-apiserver-ip-10-0-129-84.ec2.internal_kube-apiserver-6.log